### PR TITLE
fix problem with pickling thread lock #424

### PR DESCRIPTION
**Pull request by kivio**
Fixing problem with pickle thread lock's

### CODE REVIEW

**Comment by cmccoy** [comment]
Lock fix looks good - thanks! Would you mind isolating that change from the other OpenStack fixes?

File: perfkitbenchmarker/openstack/os_disk.py [hunk 1]
```diff
@@ -29,7 +28,7 @@ class OpenStackDisk(disk.BaseDisk):
 
     def __init__(self, disk_spec, name, zone, project, image=None):
         super(OpenStackDisk, self).__init__(disk_spec)
-        self.__nclient = utils.NovaClient()
+        self.__nclient = os_utils.NovaClient()
```
Comment (kivio) [inline perfkitbenchmarker/openstack/os_disk.py]: long story here

**Comment by voellm** [comment]
@kivio Marcin - Please follow the CLA bot recommendations so it does not trip on future checkins.  We value your contributions and Intel has signed the CLA but its not associating for some reason even though on GitHub you are part of the Intel group. Thanks!

**Comment by kivio** [comment]
Ok, after few troubles with CLA i'm back in action! :)

File: perfkitbenchmarker/benchmark_spec.py [hunk 2]
```diff
@@ -39,6 +42,19 @@
 from perfkitbenchmarker.rackspace import rackspace_network as rax_net
 from perfkitbenchmarker.rackspace import rackspace_virtual_machine as rax_vm
 
+
+def pickle_lock(lock):
+    return threading.Lock, (lock.locked(),)
+
+
+def unpickle_lock(locked, *args):
```
Comment (cmccoy) [inline perfkitbenchmarker/benchmark_spec.py]: Could you make these CamelCase to match style elsewhere?

**Comment by cmccoy** [comment]
Sorry @kivio, I didn't notice that you'd addressed the comment above. Thanks for this!

**Pull request merged**